pkg/gds/server.go

```diff
@@ -214,16 +217,85 @@ func (s *Server) Register(ctx context.Context, in *api.RegisterRequest) (out *ap
 		Status:     models.CertificateRequestState_INITIALIZED,
 	}
 
-	if certRequest.Pkcs12Password, certRequest.Pkcs12Signature, err = s.Encrypt(password); err != nil {
-		log.Error().Err(err).Msg("could not encrypt password to store in database")
+	// Create a new Secret Manager
+	sm, err := NewSecretManager(s.conf.Secrets, certRequest.Id)
+
+	// If the application can't access secret manager, we won't be able to keep going
+	// so raise a permissions error and return.
+	if err != nil {
+		log.Error().Err(err).Msg("unable to access secret manager")
 		out.Error = &api.Error{
-			Code:    500,
-			Message: "could not create certificate request password",
+			Code:    403,
+			Message: err.Error(),
+		}
+		return out, nil
+	}
+
+	// Make a new secret of type "password"
+	secret := "password"
+	err = sm.CreateSecret(ctx, secret)
+
+	// If the application can't create a new secret, check to see what the problem is.
+	if err != nil {
+		// If it's a timeout, that means the secret manager API call was bad
+		if errors.Is(err, context.DeadlineExceeded) {
+			log.Error().Err(err).Msg("malformed call to secret manager")
+			out.Error = &api.Error{
+				Code:    400,
+				Message: err.Error(),
+			}
+			return out, nil
+		}
+		// If the secret already exists (which means that the client has previously
+		// created a password), that's okay
+		serr, ok := status.FromError(err)
+		if ok && serr.Code() == codes.AlreadyExists {
+			log.Error().Err(err).Msg("password already exists for client; creating new password")
+		}
+		// If it's a more serious problem, that likely signals a permissions error
+		// and merits a return.
+		log.Error().Err(err).Msg("secret manager error; is GOOGLE_PROJECT_NAME set?")
+		out.Error = &api.Error{
+			Code:    403,
+			Message: err.Error(),
+		}
+		return out, nil
+	}
```

Comment:
Because this checking already happens in `CreateSecret`, I don't think it's necessary to do it again - we want to push as much responsibility to the SecretManager as possible. The only question is about what error to return for `AlreadyExists` and there are two options for how to handle that:

Option 1 - `CreateSecret` returns nil on AlreadyExists; then the check here is simply:

```go
if err = sm.CreateSecret(ctx, secret); err != nil {
    log.Error().Err(err).Msg("could not create new secret on registration")
    out.Error = &api.Error{Code: 500, Message: "internal error with registration, please contact admins")
    return out, nil
}
```

Note that the `log.Error().Err(err)` will be giving us details about the error from those `fmt.Errorf` comments I made before. We're also hiding the error from the user - we just want them to know something really bad happened and they should email us. We don't want them to know it was google secrets. Probably we should make this kind of change everywhere in the code, returning the `Message: err.Error()` is my own 🎃   code and I can create a technical debt issue for dealing with it. 

Option 2- If we want to return an `AlreadyExists` error - then we can convert the error into our own custom error like the `ErrSecretNotFound` suggestion I made earlier. The handler would then be:

```go
if err = sm.CreateSecret(ctx, secret); err != nil && !errors.Is(err, ErrSecretAlreadyExists) {
    log.Error().Err(err).Msg("could not create new secret on registration")
    out.Error = &api.Error{Code: 500, Message: "internal error with registration, please contact admins")
    return out, nil
}
```